src/duckdb.rs

```diff
@@ -0,0 +1,204 @@
+use crate::expr::{ARITHOPS, ARRAYOPS, BOOLOPS, CMPOPS, EQOPS, SPATIALOPS, TEMPORALOPS};
+use crate::Error;
+use crate::Expr;
+use crate::Geometry;
+use pg_escape::{quote_identifier, quote_literal};
+
+/// Traits for generating SQL for DuckDB with Spatial Extension
+pub trait ToDuckSQL {
+    /// Convert Expression to SQL for DuckDB with Spatial Extension
+    fn to_ducksql(&self) -> Result<String, Error>;
+}
+
+fn lit_or_prop_to_ts(arg: &Expr) -> Result<String, Error> {
+    match arg {
+        Expr::Property { property } => Ok(quote_identifier(property).to_string()),
+        Expr::Literal(v) => Ok(format!("TIMESTAMP {}", quote_literal(v))),
+        _ => Err(Error::OperationError()),
+    }
+}
+
+impl ToDuckSQL for Expr {
+    /// Converts this expression to DuckDB SQL.
+    ///
```

Comment:
@bitner maybe just throw a warning on here telling people that this function is experimental and subject to change?